test/data/config.yaml

```diff
@@ -35,7 +35,7 @@ credentials:
     ssh-key: "ssh-rsa foo"
     ssh-private-key:
         name: "id_rsa"
-        key: "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpYWFhYCi0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0tCg=="
+        key: "path/to/private/key"
```

Comment:
Shouldn't this be "$HOME/.ssh" in all cases? Since this is for storage access having a separate location is probably not supported. The NetApp doc https://library.netapp.com/ecmdocs/ECMP1155684/html/GUID-793D1333-C930-4543-9A68-02969A2E45D6.html also does not explicitly mention a special location for the private key.

It would be nice if we could just drop the "key:" option.

@jeffaco can you please confirm the location of the private key?

Reply:
I think this is a misunderstanding. path/to/private/key is the location on the lun or iso from where we fetch the private key. This can be any path. Of course the key is copied to the users home .ssh
directory. Please also look at the test which outlines this

Reply:
maybe we should change key: to fetch_key_from: or similar

Reply:
My understanding (from Operations) is that this was a DSA style key (yeah, I know that's not optimal) stored in `.ssh/id_dsa`. That's what my notes say on this.

Reply:
From my perspective, and I took another look at the code, it doesn't help us to split the name of the key and the path. I think if it is just

`ssh-private-key:
    key-location: "path/to/private/key/id_dsa`

then in the code we don't have to worry about the name at all with

`'cp', os.sep.join(
                         [ssh_key_source.location, private_key.get('key')]
                      ),  ssh_auth_dir`

Then if there is ever a switch to id_rsa the yaml code generator has only 1 place of change. Also I think the intent of this code "give me a path to a file and we'll copy it to where it belongs" is more explicitly expressed. At this point the implementation hints that the intend of the code is a rename operation, i.e. "give me a path to a file and I'll copy it to this new name in another place" Given that the file needs to be called either id_rsa or id_dsa, and this code really shouldn't care, the implication of a potential rename is what threw me for a loop.

Reply:
ok fair point and good idea, I'll change the code